lib/mixlib/install/generator/powershell/scripts/helpers.ps1.erb

```diff
@@ -136,15 +131,36 @@ function Test-ProjectPackage {
   return $Valid
 }
 
-function Get-SHA256Converter {
-  if($PSVersionTable.PSEdition -eq 'Core') {
-    [System.Security.Cryptography.SHA256]::Create()
+function Get-FileHash ($Path, $Algorithm) {
+  function disposable($o){($o -is [IDisposable]) -and (($o | get-member | foreach-object {$_.name}) -contains 'Dispose')}
+  function use($obj, [scriptblock]$sb){try {& $sb} catch [exception]{throw $_} finally {if (disposable $obj) {$obj.Dispose()}} }
+  $Path = (resolve-path $Path).providerpath
+  $hash = @{Algorithm = $Algorithm; Path = $Path}
+  use ($c = Get-SHA256Converter) {
+    use ($in = (gi $Path).OpenRead()) {
+      $hash.Hash = ([BitConverter]::ToString($c.ComputeHash($in))).Replace("-", "").ToUpper()
+    }
   }
-  else {
-    New-Object -TypeName Security.Cryptography.SHA256Managed
+  return $hash
+}
+
+function Get-SHA256Converter {
+  if (Is-FIPS) {
+    New-Object -TypeName Security.Cryptography.SHA256Cng
```

Comment:
Does this not work on some platforms? 2008, right?

It'd be nice to reduce this complexity if we could and have less options/conditionals/paths.